platform/ui/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -1,150 +1,63 @@
+  onActiveIndexChange,
   className,
-  disabled = false,
-  fullWidth = false,
-  color = 'default',
-  orientation = 'horizontal',
-  rounded = 'medium',
-  size = 'medium',
-  variant = 'outlined',
-  splitBorder = true,
-  ...other
+  orientation = ButtonEnums.orientation.horizontal,
+  defaultActiveIndex = 0,
 }) => {
-  const ref = useRef(null);
+  const [activeIndex, setActiveIndex] = useState(defaultActiveIndex);
 
-  const buttonClasses = classnames(
-    baseButtonClass,
-    borderClasses[variant] && borderClasses[variant][orientation],
-    variantClasses[variant] && variantClasses[variant][color],
-    roundedClasses[orientation] && roundedClasses[orientation][rounded]
-  );
+  const handleButtonClick = (e, index) => {
+    setActiveIndex(index);
+    onActiveIndexChange && onActiveIndexChange(index);
+    buttons[index].onClick && buttons[index].onClick(e);
+  };
+
+  const orientationClasses = {
+    horizontal: 'flex-row',
+    vertical: 'flex-col',
+  };
+
+  const wrapperClasses = classnames('inline-flex', orientationClasses[orientation], className);
 
   return (
     <div
-      role="group"
       className={classnames(
-        'group',
-        className,
-        orientationClasses[orientation],
-        fullWidth ? fullWidthDisplayClass : baseDisplayClass
+        wrapperClasses,
+        'border-secondary-light rounded-[5px] border bg-black text-[13px] '
       )}
-      ref={ref}
-      {...other}
     >
-      {React.Children.map(children, (child, index) => {
-        const isFirst = index === 0;
-        const isLast = index === React.Children.count(children) - 1;
-        const isNotFirstOrLast = !isFirst && !isLast;
-
-        if (!React.isValidElement(child)) {
-          return null;
-        }
-
-        return React.cloneElement(child, {
-          className: classnames(
-            buttonClasses,
-            child.props.className,
-            !splitBorder && isNotFirstOrLast && nonFirstLastClasses[orientation],
-            !splitBorder && isLast && 'last:border-l-0'
-          ),
-          disabled: child.props.disabled || disabled,
-          color: child.props.color || color,
-          fullWidth,
-          rounded: 'none',
-          size: child.props.size || size,
-          variant: child.props.variant || variant,
-        });
+      {buttons.map((buttonProps, index) => {
+        const isActive = index === activeIndex;
+        return (
+          <button
+            {...buttonProps}
+            key={index}
+            className={classnames(
+              'rounded-[4px] px-2 py-1',
+              isActive ? 'bg-customblue-40 text-white' : 'text-primary-active bg-black'
+            )}
+            onClick={e => handleButtonClick(e, index)}
+          />
+        );
       })}
     </div>
   );
 };
 
 ButtonGroup.propTypes = {
-  children: PropTypes.node.isRequired,
+  buttons: PropTypes.arrayOf(PropTypes.object).isRequired,
+  orientation: PropTypes.oneOf(['horizontal', 'vertical']),
```

Comment:
To avoid using strings, consider declaring this as you did below`PropTypes.oneOf(Object.values(ButtonEnums.orientation))`.